### PR TITLE
upgrading to discord.net 2.4.0

### PR DESCRIPTION
**Pull request by confessore**
let me know if I need to make any additional edits

### CODE REVIEW

File: Discord.Net.Abstractions/Discord.Net.Abstractions.csproj [hunk 1]
```diff
@@ -5,7 +5,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Discord.Net" />
+    <PackageReference Include="Discord.Net" Version="2.4.0" />
```
Comment (JakenVeina) [inline Discord.Net.Abstractions/Discord.Net.Abstractions.csproj:8]: Revert this.

File: Discord.Net.Abstractions/Rest/IDiscordRestClient.cs [hunk 2]
```diff
@@ -120,6 +120,7 @@ public async Task<IRestGuild> GetGuildAsync(ulong id, RequestOptions options = n
                 ?.Abstract();
 
         /// <inheritdoc />
+        [Obsolete("This endpoint is deprecated, please use GetGuildWidgetAsync instead.")]
```
Comment (JakenVeina) [inline Discord.Net.Abstractions/Rest/IDiscordRestClient.cs:123]: This matches an annotation added by Discord.NET?

File: Modix.Data/Modix.Data.csproj [hunk 3]
```diff
@@ -4,7 +4,7 @@
     <Nullable>enable</Nullable>
   </PropertyGroup>
   <ItemGroup>
-    <PackageReference Include="Discord.Net" />
+    <PackageReference Include="Discord.Net" Version="2.4.0" />
```
Comment (JakenVeina) [inline Modix.Data/Modix.Data.csproj:7]: Revert this.

File: Modix.Services/Modix.Services.csproj [hunk 4]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <ItemGroup>
     <PackageReference Include="AsyncEvent" />
-    <PackageReference Include="Discord.Net" />
+    <PackageReference Include="Discord.Net" Version="2.4.0" />
```
Comment (JakenVeina) [inline Modix.Services/Modix.Services.csproj:4]: Revert this.

File: Modix.Services/BehaviorBase.cs [hunk 5]
```diff
@@ -89,7 +89,7 @@ private bool _hasDisposed
         /// </summary>
         /// <param name="action">The action to be executed.</param>
         /// <returns>A <see cref="Task"/> that will complete when the operation has completed.</returns>
-        internal protected async Task SelfExecuteRequest(Func<IServiceProvider, Task> action)
+        internal protected async Task SelfExecuteRequestAsync(Func<IServiceProvider, Task> action)
```
Comment (JakenVeina) [inline Modix.Services/BehaviorBase.cs:92]: Not that I don't agree with this change, but it's out-of-scope.

File: Modix.sln [hunk 6]
```diff
@@ -15,11 +15,17 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Modix.Data.Test", "Modix.Da
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution Items", "{23DA774D-7AE9-48C1-A261-F27D15A07858}"
 	ProjectSection(SolutionItems) = preProject
+		.dockerignore = .dockerignore
```
Comment (JakenVeina) [inline Modix.sln:18]: Not that I don't agree with these changes, but they're out-of-scope.

File: Modix/Modix.csproj [hunk 7]
```diff
@@ -19,6 +19,7 @@
 
   <ItemGroup>
     <PackageReference Include="AspNet.Security.OAuth.Discord" />
+    <PackageReference Include="Discord.Net" Version="2.4.0" />
```
Comment (JakenVeina) [inline Modix/Modix.csproj:22]: Remove the version specification. Also, probably remove this entirely. It hasn't been needed before, why would it be needed now?

**Review by JakenVeina — CHANGES_REQUESTED** [review]

File: Discord.Net.Abstractions/Rest/IDiscordRestClient.cs (hunk 2 above)
Comment (confessore) [reply Discord.Net.Abstractions/Rest/IDiscordRestClient.cs:123]: correct. this matches an annotation added by discord.net

**Review by JakenVeina — APPROVED** [review]

**Review by Scott-Caldwell — APPROVED** [review]
Pulled the changes and tested locally last night, looked like everything was still working after the upgrade

**Pull request merged**